osu.Game/Screens/SelectV2/SongSelect.cs

```diff
@@ -756,7 +756,9 @@ protected override void LogoArriving(OsuLogo logo, bool resuming)
 
             logo.Action = () =>
             {
-                SelectAndRun(Beatmap.Value.BeatmapInfo, OnStart);
+                // use carousel's current selection instead of global beatmap to avoid debounce issues
+                var beatmapToPlay = carousel.CurrentGroupedBeatmap?.Beatmap ?? Beatmap.Value.BeatmapInfo;
```

Comment:
First of all, there's already an established way of doing this sort of thing, which is to call

```suggestion
                ensureGlobalBeatmapValid();
```

which will work to fix the issue because it has the following call inside:

https://github.com/ppy/osu/blob/766334ae34a32d69716ec419348a156b8568037e/osu.Game/Screens/SelectV2/SongSelect.cs#L577

and will *additionally* run several sanity checks that this cowboy direct access into the carousel will not.

Secondly, this is likely a relevant concern in another call site of `SelectAndRun()`, namely

https://github.com/ppy/osu/blob/0b3ec3f1e1982a77355c69298e5a292c9f24ece2/osu.Game/Screens/SelectV2/SongSelect.cs#L997-L1003

which should probably receive similar treatment.

Thirdly (and optionally), it'd be nice to see test coverage for this - although with this being a timing issue I could see it being finicky to spend time writing tests for so I would not spend more than an hour or two on it.